src/services/auth.service.js

```
@@ -0,0 +1,203 @@
+
+const register = async ({ name, email, password }) => {
+  const errors = {
+    name: checkRequired(name, 'name'),
+    email: validateEmail(email),
+    password: validatePassword(password),
+  };
+
+  if (errors.email || errors.name || errors.password) {
+    throw ApiError.badRequest('bad request', errors);
+  }
+
+  if (await userService.getByEmail(email)) {
+    throw ApiError.badRequest('bad request', {
+      userExists: 'user already exists',
+    });
+  }
+
+  const hashedPass = await bcrypt.hash(password, 10);
+  const activationToken = uuidv4();
+  const newUser = await User.create({
+    name,
+    email,
+    password: hashedPass,
+    activationToken,
+  });
+
+  await emailService.sendActivationEmail(email, activationToken);
+
+  return newUser;
+};
+
+const activate = async (activationToken) => {
+  const user = await userService.getByActivationToken(activationToken);
+
+  if (!user) {
+    throw ApiError.notFound();
+  }
+
+  user.activationToken = null;
+  await user.save();
+
+  return user;
+};
+
+const generateTokens = async (user) => {
+  const normalizedUser = userService.normalize(user);
+
+  const accessToken = jwtService.sign(normalizedUser);
+  const refreshToken = jwtService.signRefresh(normalizedUser);
+
+  await tokenService.save(normalizedUser.id, refreshToken);
+
+  return { user: normalizedUser, refreshToken, accessToken };
+};
+
+const login = async ({ email, password }) => {
+  const errors = {
+    email: checkRequired(email, 'email'),
+    password: checkRequired(password, 'password'),
+  };
+
+  if (errors.email || errors.password) {
+    throw ApiError.badRequest('bad request', errors);
+  }
+
+  const user = await userService.getByEmail(email);
+
+  if (!user) {
+    throw ApiError.badRequest('no such user');
+  }
+
+  const isPasswordValid = await bcrypt.compare(password, user.password);
+
+  if (!isPasswordValid) {
+    throw ApiError.badRequest('wrong password');
+  }
+
+  return generateTokens(user);
```

The task description requires that a user cannot log in if their account is not yet activated. This function successfully validates the user's credentials but does not check if the user's account is active (i.e., if `user.activationToken` is not `null`). You should add a check after finding the user and before generating tokens to ensure they are activated. If not, you should throw an error instructing them to check their email.